pkg/controllermanager/controller/shoot/shoot_control.go

```diff
@@ -200,160 +149,163 @@ func (c *Controller) updateShootStatusProcessing(shoot *gardenv1beta1.Shoot, mes
+				o.Logger.Error(lastErr.Description)
+				if err := c.updateShootStatusReconcileError(o, operationType, lastErr); err != nil {
+					o.Logger.WithError(err).Error("Could not update the Shoot status after reconciliation error")
+					return reconcile.Result{}, err
+				}
+			}
+			return reconcile.Result{}, err
 		}
+	}
 
-		shootLogger.Infof("Successfully rescheduled failed shoot %q for reconciliation due to Gardener version update", shoot.Name)
+	if common.IsShootFailed(shoot) || common.ShouldIgnoreShoot(c.respectSyncPeriodOverwrite(), shoot) {
+		o.Logger.Info("Shoot is failed or ignored")
+		return reconcile.Result{}, nil
 	}
 
-	// When a Shoot clusters deletion timestamp is set we need to delete the cluster and must not trigger a new reconciliation operation.
-	if shoot.DeletionTimestamp != nil {
-		c.recorder.Eventf(shoot, corev1.EventTypeNormal, gardenv1beta1.EventDeleting, "[%s] Deleting Shoot cluster", operationID)
-		if updateErr := c.updateShootStatusDeleteStart(operation); updateErr != nil {
-			shootLogger.Errorf("Could not update the Shoot status after deletion start: %+v", updateErr)
-			return true, updateErr
-		}
-		if deleteErr := c.deleteShoot(operation); deleteErr != nil {
-			shootLogger.Errorf("Error during execution of the delete operation for shoot: %+v", deleteErr.Description)
-			c.recorder.Eventf(shoot, corev1.EventTypeWarning, gardenv1beta1.EventDeleteError, "[%s] %s", operationID, deleteErr.Description)
-			if state, updateErr := c.updateShootStatusDeleteError(operation, deleteErr); updateErr != nil {
-				shootLogger.Errorf("Could not update the Shoot status after deletion error: %+v", updateErr)
-				return state != gardencorev1alpha1.LastOperationStateFailed, updateErr
-			}
-			return true, errors.New(deleteErr.Description)
-		}
-		if err := operation.DeleteClusterResourceFromSeed(context.TODO()); err != nil {
-			lastErr := &gardencorev1alpha1.LastError{Description: fmt.Sprintf("Could not delete Cluster resource in seed: %s", err)}
-			c.recorder.Eventf(shoot, corev1.EventTypeWarning, gardenv1beta1.EventDeleteError, "[%s] %s", operationID, lastErr.Description)
-			if state, updateErr := c.updateShootStatusDeleteError(operation, lastErr); updateErr != nil {
-				shootLogger.Errorf("Could not update the Shoot status after deletion error: %+v", updateErr)
-				return state != gardencorev1alpha1.LastOperationStateFailed, updateErr
-			}
-			return true, errors.New(lastErr.Description)
-		}
-		c.recorder.Eventf(shoot, corev1.EventTypeNormal, gardenv1beta1.EventDeleted, "[%s] Deleted Shoot cluster", operationID)
-		if updateErr := c.updateShootStatusDeleteSuccess(operation); updateErr != nil {
-			shootLogger.Errorf("Could not update the Shoot status after deletion success: %+v", updateErr)
-			return true, updateErr
+	if c.reconcileInMaintenanceOnly() && (!common.IsOperationOngoing(shoot) && !common.IsNowInEffectiveShootMaintenanceTimeWindow(shoot)) {
```

Comment:
When  getting a new gardener identity (locally restarted the controller manager, new Image Version, ..) it will always reconcile all the shoots. Even though setting reconcileOnlyInMaintenanceOnly == true  is set & we are not in the MaintenanceTime Window.
This is the case because there is an operationOngoing / lastOperation.State != LastOperationStateSucceeded .  --> shoot directly after restart (until seed is reconciled) is not on a healthy seed --> this sets a condition on the shoot

Should all the seeds and after that, all the shoots be reconciled on a new gardener release deployment?  As a user, I would expect the shoot to only reconcile in the MaintenanceTime Window when the flag is explicitly set.
What do you think?

Reply:
As @rfranzke suggested, I recently also introduced a flag that made this 'reconcile during maintenance time window' configurable and turned off by default. Did you check with that change / did you enable `reconcileDuringMaintenanceOnly`?

Reply:
Yes the flag was set to "true". However it reconciles anyways, thats what I was trying to say.

Reply:
Very good catch! I'm now not updating the status on the shoot anymore if I'm 'not allowed to', considering that the Seed must've been healthy in the past (otherwise a reconciliation wouldn't have gone through). This resolves that issue and makes sure that we only update during the maintenance time window.